BrainPortal/lib/cbrain_task_generators/templates/bourreau.rb.erb

```diff
@@ -267,10 +298,42 @@ class CbrainTask::<%= name %> < <%= descriptor['cbrain:inherits-from-class'] ||
 %     end
     })
 
+%#  If we are going to execute in a container, we need to detect output files outside the mounted directory,
+%#  add commands to mv them there, and then change the file paths to look there when saving results.
+%   containerImage = descriptor['container-image']
+    outfileMoveCommands = []
+%   if containerImage
+    # If execution will occur in a container, alter the output file paths to compensate
+    if self.use_docker?
+      # Task launch directory (mounted into as pwd if specified)
+      # Note: assumes container_wd is absolute
+%     customWd = containerImage['working-directory']
+      container_wd = <%= customWd ? "\"" + customWd + "\"" : "self.full_cluster_workdir" %>
+%     outids = outputs.map { |p| ":'#{p['id']}'" }.join(", ")
+      is_underneath = lambda { |base,potential| (Pathname.new(potential)).fnmatch?(File.join(base,'**')) }
+      [<%= outids %>].each do |outid|
+        # Do not issue move commands for optional output files with empty path templates
+        next if params[ outid ].nil? or params[ outid ] == ''
+        # Relative paths are assumed to be with respect to the mounted directory
+        currentPath = File.absolute_path( params[ outid ], container_wd )
+        # If the current path is not in the directory subtree of the mounted directory,
+        # issue a move command at the end of cluster_commands and alter the path to search
+        # for output files accordingly
+        unless is_underneath.( container_wd, currentPath )
+          # Move the file to the mounted dir (to be run post-execution, in the container)
+          # Warning: potential overwrite dangers if there is a same-named local file
+          newPosition = File.join( container_wd, File.basename(currentPath) )
+          outfileMoveCommands << "mv #{currentPath} #{newPosition}"
```

Comment:
Will these `mv` command properly escape path with special characters in them? Filenames with spaces, hash signs, at signs, dollar signs, backticks, UTF8-encoded characters?